rfcs/01780_policy_delegation.md

```diff
@@ -0,0 +1,326 @@
+identified by `https://google.com/oak/keys/count_people_from_pixels.pub` to sign
+hashes of modules that count the number of cats instead of the number of people.
+The only understanding of what the hashes signed with a given particular key
+"mean" is outside of the key signing system itself. The owner of the private key
+may make formal or informal guarantees about functionality or policies related
+to the trusted objects, but those need to be published separately.
+
+For instance, Google may have a separate web page in which expectations of what
+is signed with each private key are detailed, with links to corresponding public
+key, and, if relevant, perhaps also technical instructions for third parties to
+independently verify that such properties are upheld (e.g. source code, training
+data, etc.). None of this affects the functioning of the delegation mechanism in
+Oak though.
+
+## Information Flow Control and Declassification Privilege
+
+Oak is extended with a new principal referring to manifests. When a piece of
+data is labelled with the URL of a manifest, Oak treats it as a regular IFC
+label.
+
+The Oak Runtime is extended to accept a list of URLs of initial Discovery
+Manifests which it recursively expands (up to a configurable recursion limit,
+and avoiding cycles) at startup, creating a local cache mapping each public key
+to the list of modules signed with it, and each module to the list of keys that
+signed it.
+
+When an Oak Application creates a new Wasm Node, the Oak Runtime checks the hash
+of the Wasm module to instantiate, and if it corresponds to one or more valid
+signed module hashes, then it grants the newly created node the privilege
+corresponding to their respective public keys.
+
+For instance, let us assume the following configuration:
+
+- `https://google.com/oak/keys/count_people_from_pixels.pub`:
+
+  - contents:
+
+    ```text
+    -----BEGIN PUBLIC KEY-----
+    aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+    -----END PUBLIC KEY-----
+    ```
+
+  - used to sign modules with the following hashes:
+    - `0000000000000000000000000000000000000000`
+    - `1111111111111111111111111111111111111111`
+
+- `https://other-company.com/oak/keys/count_people_from_pixels.pub`:
+
+  - contents:
+
+    ```text
+    -----BEGIN PUBLIC KEY-----
+    bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
+    -----END PUBLIC KEY-----
+    ```
+
+  - used to sign modules with the following hashes:
+    - `1111111111111111111111111111111111111111`
+    - `2222222222222222222222222222222222222222`
+
+When an Oak Node with hash `1111111111111111111111111111111111111111` is
+instantiated, it receives the declassification privilege of
+"signature:aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa ∧
+signature:bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb ∧
+wasm_hash:1111111111111111111111111111111111111111".
+
+Note that in this case it receives both privileges (or rather, the conjunction
```

Comment:
I am a bit confused. What are the privileges? Are they declassifying content from `google.com` and `other-company.com` (based on the `aa...aa` and `bb...bb` keys)?